pkg/util/errutil/status.go

```diff
@@ -110,6 +123,16 @@ func (s CoreStatus) LogLevel() LogLevel {
 	}
 }
 
+// IsDownstream returns if CoreStatus is originating from a downstream service.
+func (s CoreStatus) IsDownstream() bool {
+	switch s {
+	case StatusBadGateway, StatusGatewayTimeout:
```

Comment:
Are those the only 5xx status codes we return if an datasource is having problems?

This could work for datasources but the idea for this label is that it should also work for ex alerting that sends requests to the alertmanager.

But perhaps we set those downstream errors "manually" instead.

Reply:
Earlier we've always used 500 when receiving an error from backend plugin. My idea is that 502 bad gateway make more sense. 

> alerting that sends requests to the alertmanager.

Couldn't they use `errutil.BadGateway` as well?

> But perhaps we set those downstream errors "manually" instead.

This was the obvious idea for plugins as well, but felt wrong adding requestmeta dependency to the plugin package

Reply:
> Couldn't they use errutil.BadGateway as well?

yea that seems reasonable! If not and their client needs to stick to current status codes they could add it to the context manually.

Reply:
I just pushed some changes that allows to denote an errutil.Base with downstream source, if needed
https://github.com/grafana/grafana/blob/ca7cac785e771b3ac53fe77286d5e65675dd9995/pkg/util/errutil/errors.go#L198-L206C2